k8s-install/scripts/install-cni.sh

```diff
@@ -48,45 +48,35 @@ then
 	CNI_CONF_ETCD_CERT=${HOST_SECRETS_DIR}/etcd-cert
 fi
 
+# Choose which default cni binaries should be copied
+SKIP_CNI_BINARIES=${SKIP_CNI_BINARIES:-""}
+SKIP_CNI_BINARIES=",$SKIP_CNI_BINARIES,"
+
 # Place the new binaries if the directory is writeable.
 if [ -w "/host/opt/cni/bin/" ]; then
-	cp /opt/cni/bin/calico /host/opt/cni/bin/
-	cp /opt/cni/bin/calico-ipam /host/opt/cni/bin/
-	# Copy over the 3rd party CNI binaries, but do not clobber if they exist
-	if [ ! -f /host/opt/cni/bin/flannel ]; then
-	    cp /opt/cni/bin/flannel /host/opt/cni/bin/
-	fi
-	if [ ! -f /host/opt/cni/bin/loopback ]; then
-	    cp /opt/cni/bin/loopback /host/opt/cni/bin/
-	fi
-	if [ ! -f /host/opt/cni/bin/host-local ]; then
-	    cp /opt/cni/bin/host-local /host/opt/cni/bin/
-	fi
-	if [ ! -f /host/opt/cni/bin/portmap ]; then
-	    cp /opt/cni/bin/portmap /host/opt/cni/bin/
-	fi
+	for path in /opt/cni/bin/*
+	do
+		filename=$(basename $path)
+		tmp=",$filename,"
+		if [ ! "${SKIP_CNI_BINARIES#*$tmp}" != "$SKIP_CNI_BINARIES" ] && [ ! -f /host/opt/cni/bin/$filename ]; then
```

Comment:
Is there a reason for the double negative? `[ ! x != y ]`  If not please switch to equality check, otherwise please add a comment.